Month3/PixelGameEngine/AsteroidGame/AsteroidMain.cpp

```diff
@@ -0,0 +1,463 @@
+#define OLC_PGE_APPLICATION
+#include <algorithm>
+#include "olcPixelGameEngine.h"
+
+
+class AsteroidGame : public olc::PixelGameEngine
+{
+public:
+	AsteroidGame()
+	{
+		sAppName = "Asteroids";
+	}
+private:
+
+	struct SpaceObject
+	{
+		float PosX;
+		float PosY;
+		float DisplacementX;
+		float DisplacementY;
+		int Size;
+		float Angle;
+		bool BeDead;
+	};
+	std::vector <SpaceObject> VecAsteroid;
+	std::vector<SpaceObject> Bullets;
+	std::vector<SpaceObject> NewAsteroids;
+
+	SpaceObject Player;
+
+	std::vector<std::pair<float, float>> ModelShip;
+	std::vector<std::pair<float, float>> ModelAsteroids;
+	int Level = 1;
+	int Score = 0;
+
+protected:
+
+	void WrapCoordinates(float CoordinatesX, float CoordinatesY, float& WrapedX, float& WrapedY)
+	{
+		WrapedX = CoordinatesX;
+		WrapedY = CoordinatesY;
+		if (CoordinatesX < 0.0f)
+		{
+			WrapedX = CoordinatesX + (float)ScreenWidth();
+		}
+		if (CoordinatesX > ScreenWidth())
+		{
+			WrapedX = CoordinatesX - (float)ScreenWidth();
+		}
+
+		if (CoordinatesY < 0.0f)
+		{
+			WrapedY = CoordinatesY + (float)ScreenHeight();
+		}
+		if (CoordinatesY > ScreenHeight())
+		{
+			WrapedY = CoordinatesY - (float)ScreenHeight();
+		}
+	}
+
+	void BackgroundColor()
```

Comment:
you don need to create this with a code

Reply:
furthermore according to our coding standards, the methods have to be named with a verb